internal/dockercompose/wrapper.go

```diff
@@ -0,0 +1,135 @@
+package dockercompose
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/windmilleng/tilt/internal/logger"
+)
+
+type DockerComposeClient interface {
+	Up(ctx context.Context, pathToConfig, serviceName string, stdout, stderr io.Writer) error
+	Down(ctx context.Context, pathToConfig string, stdout, stderr io.Writer) error
+	Logs(ctx context.Context, pathToConfig, serviceName string) (io.ReadCloser, error)
+	Events(ctx context.Context, pathToConfig string) (<-chan string, error)
+	Config(ctx context.Context, pathToConfig string) (string, error)
+	Services(ctx context.Context, pathToConfig string) (string, error)
+}
+
+type cmdDCClient struct{}
+
+func NewDockerComposeClient() DockerComposeClient {
+	return &cmdDCClient{}
+}
+
+func (c *cmdDCClient) Up(ctx context.Context, pathToConfig, serviceName string, stdout, stderr io.Writer) error {
+	cmd := exec.CommandContext(ctx, "docker-compose", "-f", pathToConfig, "up", "--no-deps", "--build", "--force-recreate", "-d", serviceName)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	err := cmd.Run()
+	return FormatError(cmd, nil, err)
+}
+
+func (c *cmdDCClient) Down(ctx context.Context, pathToConfig string, stdout, stderr io.Writer) error {
+	cmd := exec.CommandContext(ctx, "docker-compose", "-f", pathToConfig, "down")
+	cmd.Stdout = logger.Get(ctx).Writer(logger.InfoLvl)
+	cmd.Stderr = logger.Get(ctx).Writer(logger.InfoLvl)
+
+	err := cmd.Run()
+	if err != nil {
+		return FormatError(cmd, nil, err)
+	}
+
+	return nil
+}
+
+func (c *cmdDCClient) Logs(ctx context.Context, pathToConfig, serviceName string) (io.ReadCloser, error) {
```

Comment:
hmm maybe worth a comment that this keeps running/tailing logs? (or a rename to indicate?)